File: examples/greeter/js/client/greeter_service_stub.js
const {
    marshalRequest,
    unmarshalResponse,
    sendRequest,
    DEFAULT_TIMEOUT,
    GREETER_SERVICE,
    SAY_HELLO_RPC,
    SAY_BYE_RPC
} = require('./client');

async function sayHello(request, headers = {}) {
    const binReq = marshalRequest(request);
    const [respBody, respHeaders] = await sendRequest(greeterAddr, GREETER_SERVICE, SAY_HELLO_RPC, binReq, headers, greeterTimeout);
    return unmarshalResponse(respBody, respHeaders, SAY_HELLO_RPC);
}

async function sayBye(request, headers = {}) {
    const binReq = marshalRequest(request);
    const [respBody, respHeaders] = await sendRequest(greeterAddr, GREETER_SERVICE, SAY_BYE_RPC, binReq, headers, greeterTimeout);
    return unmarshalResponse(respBody, respHeaders, SAY_BYE_RPC);
}

const greeterAddr = process.env.GREETER_SERVICE_ADDR;
if (!greeterAddr) {
    console.error("GREETER_SERVICE_ADDR environment variable not set");
    throw new Error("GREETER_SERVICE_ADDR environment variable not set");
}

let t = process.env.GREETER_SERVICE_TIMEOUT;
if (!t) {
    t = DEFAULT_TIMEOUT;
} else {
    t = parseInt(t, 10);
    if (isNaN(t) || t <= 0) {
        t = DEFAULT_TIMEOUT;
    }
}
const greeterTimeout = t

module.exports = {sayHello, sayBye};

File: examples/grpc_service/python/client/greeter_service_stub.py
import os

from client import marshal_request, unmarshal_response, send_request, DEFAULT_TIMEOUT, \
    GREETER_SERVICE, SAY_HELLO_RPC, SAY_BYE_RPC
from logger import get_simple_logger

logger = get_simple_logger('greeterservice-stub')


def say_hello(request, headers=None):
    bin_req = marshal_request(request)
    resp_body, headers = send_request(greeter_addr, GREETER_SERVICE, SAY_HELLO_RPC, bin_req, headers, greeter_timeout)
    return unmarshal_response(resp_body, headers, SAY_HELLO_RPC)


def say_bye(request, headers=None):
    bin_req = marshal_request(request)
    resp_body, headers = send_request(greeter_addr, GREETER_SERVICE, SAY_BYE_RPC, bin_req, headers, greeter_timeout)
    return unmarshal_response(resp_body, headers, SAY_BYE_RPC)


greeter_addr = os.getenv("GREETER_SERVICE_ADDR")
if greeter_addr is None:
    logger.error("GREETER_SERVICE_ADDR environment variable not set")
    raise EnvironmentError("GREETER_SERVICE_ADDR environment variable not set")

t = os.getenv("GREETER_SERVICE_TIMEOUT")
if t is None:
    greeter_timeout = DEFAULT_TIMEOUT
else:
    try:
        greeter_timeout = int(t)
        if greeter_timeout <= 0:
            greeter_timeout = DEFAULT_TIMEOUT
    except ValueError:
        greeter_timeout = DEFAULT_TIMEOUT
